hipcheck/src/analysis/report_builder.rs

```diff
@@ -28,20 +28,26 @@ pub fn build_report(session: &Session, scoring: &ScoringResults) -> Result<Repor
 
 	// Activity analysis.
 
-	match &scoring.results.activity {
-		Some((Ok(pred), concerns)) => {
-			let HCBasicValue::Unsigned(value) = pred.value else {
-				return Err(hc_error!("activity analysis has a non-u64 value"));
-			};
-			let HCBasicValue::Unsigned(thresh) = pred.threshold else {
-				return Err(hc_error!("activity analysis has a non-u64 value"));
-			};
-			builder.add_analysis(Analysis::activity(value, thresh), concerns.clone())?;
-		}
-		Some((Err(error), _concerns)) => {
-			builder.add_errored_analysis(AnalysisIdent::Activity, error);
+	if let Some(stored) = &scoring.results.activity {
+		match &stored.result {
+			Ok(analysis) => {
+				let Some(pred) = analysis.as_any().downcast_ref::<ThresholdPredicate>() else {
```

Comment:
Maybe I've missed something, it looks like you have a `ThresholdPredicate`, use `as_any` to get an `Any` trait object, then immediately downcast it back to the original type. Why the maneuver?

Reply:
I have a `Rc<dyn HCPredicate>`, not a `Rc<ThresholdPredicate>`, so the type is not known at compile-time.

I changed `AnalysisResults` to store `HCStoredResults` instead of `ThresholdPredicate` for consistency with the new `AltAnalysisResult` functionality. It cleans up the code in `score.rs` further.

Reply:
If I try something like this:
```
let p: Rc<dyn HCPredicate> = analysis.clone();
let x = p.downcast::<ThresholdPredicate>();
```

I get `method not found in Rc<dyn HCPredicate>` even though it extends `Any`